### PR TITLE
dtc: 1.6.1 -> 1.7.0

### PR DESCRIPTION
**Pull request by otavio**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by otavio** [comment]
Result of `nixpkgs-review pr 234527` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>argononed</li>
    <li>bareboxTools</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>diffoscopeMinimal</li>
    <li>diffoscopeMinimal.dist</li>
    <li>diffoscopeMinimal.man</li>
    <li>dt-schema (python310Packages.dtschema)</li>
    <li>dt-schema.dist (python310Packages.dtschema.dist)</li>
    <li>dtc</li>
    <li>fwts</li>
    <li>python310Packages.libfdt</li>
    <li>python311Packages.dtschema</li>
    <li>python311Packages.dtschema.dist</li>
    <li>python311Packages.libfdt</li>
    <li>restool</li>
    <li>spike</li>
    <li>sunxi-tools</li>
    <li>ubootQemuX86</li>
    <li>ubootTools</li>
  </ul>
</details>

File: pkgs/development/compilers/dtc/default.nix [hunk 1]
```diff
@@ -14,44 +16,61 @@
...
-    ./0001-Depend-on-.a-instead-of-.so-when-building-static.patch
+    # meson.build: bump version to 1.7.0
+    (fetchpatch {
+      url = "https://github.com/dgibson/dtc/commit/64a907f08b9bedd89833c1eee674148cff2343c6.patch";
+      sha256 = "sha256-p2KGS5GW+3uIPgXfuIx6aDC54csM+5FZDkK03t58AL8=";
+    })
+
+    # Fix version in libfdt/meson.build
+    (fetchpatch {
+      url = "https://github.com/dgibson/dtc/pull/95/commits/723545ebe9933b90ea58dc125e4987c6bcb04ade.patch";
```
Comment (AndersonTorres) [inline pkgs/development/compilers/dtc/default.nix:42]: Why some are in /pull but others in /commit?

File: pkgs/development/compilers/dtc/default.nix [hunk 2]
```diff
@@ -14,44 +16,61 @@
...
+    # Use #ifdef NO_VALGRIND
+    (fetchpatch {
+      url = "https://github.com/dgibson/dtc/commit/41821821101ad8a9f83746b96b163e5bcbdbe804.patch";
+      sha256 = "sha256-7QEFDtap2DWbUGqtyT/RgJZJFldKB8oSubKiCtLZ0w4=";
+    })
   ];
 
-  nativeBuildInputs = [ flex bison pkg-config which ]
-    ++ lib.optionals pythonSupport [ python swig ];
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```
Comment (AndersonTorres) [inline pkgs/development/compilers/dtc/default.nix:53]:
```suggestion
  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

File: pkgs/development/compilers/dtc/default.nix [hunk 3]
```diff
@@ -14,44 +16,61 @@
...
+
+  nativeBuildInputs = [ meson ninja flex bison pkg-config which ]
```
Comment (AndersonTorres) [inline pkgs/development/compilers/dtc/default.nix:55]: One per line.

File: pkgs/development/compilers/dtc/default.nix [hunk 4]
```diff
@@ -14,44 +16,61 @@
...
 
   postPatch = ''
-    patchShebangs pylibfdt/
+    patchShebangs setup.py
   '';
 
-  makeFlags = [ "PYTHON=python" "STATIC_BUILD=${toString stdenv.hostPlatform.isStatic}" ];
-  installFlags = [ "INSTALL=install" "PREFIX=$(out)" "SETUP_PREFIX=$(out)" ];
+  # Required for installation of Python library and is innocuous otherwise.
+  DESTDIR = "/";
```
Comment (AndersonTorres) [inline pkgs/development/compilers/dtc/default.nix:65]:
```suggestion
  env.DESTDIR = "/";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/dtc/default.nix (hunk 1 above)
Comment (otavio) [reply pkgs/development/compilers/dtc/default.nix:42]: This has not been merged; others are.

Comment (AndersonTorres) [reply pkgs/development/compilers/dtc/default.nix:42]: When I enter 

https://github.com/dgibson/dtc/pull/95/commits/723545ebe9933b90ea58dc125e4987c6bcb04ade.patch

it redirects to

https://github.com/dgibson/dtc/commit/723545ebe9933b90ea58dc125e4987c6bcb04ade.patch

No real difference.

Comment (otavio) [reply pkgs/development/compilers/dtc/default.nix]: Fixed.

**Comment by otavio** [comment]
@vcunat where? I didn't see the log.

**Comment by vcunat** [comment]
CI gets collapsed after merging.  You can expand the list by clicking "View details" button.

**Comment by vcunat** [comment]
After merging it also appears on the build farm:
https://hydra.nixos.org/eval/1795705#tabs-now-fail

**Pull request merged**